### PR TITLE
Add `find` command

### LINKED ISSUE #24: find recipes using my limited ingredients

### PR DESCRIPTION
**Pull request by Sukkaito**
This PR resolves #24, with some additions on top
### Feature:
- `find` command: `find [<options>/] <query>`
  + `options` could be any combinations of "nNiIsS" with max length of 3, ends with a forward slash.

### Changes:
- Add `printFind()` to `UI.java`.
- Extract `printListwithOrder()` to a separate method for more flexibility.
- test: JUnit exception expected tests for find command

### Todo list:
- Edge cases for `options` (e.g: `find nNn/nut`)
- JUnit tests for success runs

### CODE REVIEW

File: src/main/java/ymfc/commands/FindCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public void execute(RecipeList recipes, Ui ui, Storage storage) throws Exception {
+        logger.log(Level.FINEST, "Executing FindCommand");
+        ArrayList<Recipe> results = recipes.getRecipes()
+                .stream()
+                .filter((Recipe recipe) -> isQueryFoundInName(recipe) |
+                        isFoundInIngredients(recipe) |
+                        isFoundInSteps(recipe)
+                )
+                .collect(Collectors.toCollection(ArrayList::new));
```
Comment (seanngja) [inline src/main/java/ymfc/commands/FindCommand.java:51]: Nice use of a stream here to extract out the recipes that match

**Review by seanngja — APPROVED** [review]
LGTM!
